eth/eventsyncer/event_syncer.go

```diff
@@ -110,6 +130,13 @@ func (es *EventSyncer) SyncHistory(ctx context.Context, fromBlock uint64) (lastP
 		return 0, fmt.Errorf("event replay: lastProcessedBlock (%d) is lower than fromBlock (%d)", lastProcessedBlock, fromBlock)
 	}
 
+	// Check if the block is too old.
+	// #nosec G115
+	b := big.NewInt(int64(es.lastProcessedBlock))
```

Comment:
I get the impression that this is safer, and you don't need to convert uint to int.
```suggestion
	b := new(big.Int).SetUint64(es.lastProcessedBlock)
```